target/scripts/helpers/error.sh

```diff
@@ -25,7 +25,6 @@ function dms_panic
   local PANIC_TYPE=${1:-}
   local PANIC_INFO=${2:-}
   local PANIC_SCOPE=${3-} # optional, must not be :- but just -
```

Comment:
@georglauterbach 

Can you explain why `${3-}` must be used? `${3:-}` should do the same here.

Reply:
With your addition, both are actually fine. `${3-}` was required before because the (non-)nullity check that Bash does when using `:-` (instead of `-`) would lead to the situation in which `''` is treated as `null`, so `'' 'second_argument'` would become `'second_argument'`, which is not what we wanted.

Because now there is no 4th argument, we can make it `:-` again.

Reply:
```suggestion
  local PANIC_SCOPE=${3:-}
```

Reply:
Sorry, I was looking at it from the wrong side:

```console
$ function foo() { local ONE=${1:-lol} ; local TWO=${2:-} ; echo "|$ONE|$TWO|" ; }
$ foo '' 2
|lol|2|
$ function foo() { local ONE=${1-lol} ; local TWO=${2:-} ; echo "|$ONE|$TWO|" ; }
$ foo '' 2
||2|
```

Seems like that for our example, it really made no difference to begin with as we're assigning the empty string anyways.